### ISSUE TITLE
Error : Could not extract users from ldap

### ISSUE THREAD

**Issue by kal-u** [Processus-Thief/HEKATOMB#2]
Bonjour Proc,

ci-dessous un retour et une limite de ton outil. 
Dans un domaine avec des milliers d'utilisateurs, on obtient le message d'erreur suivant :

Testing admin rights...
Admin access granted.
Testing LDAP connection...
LDAP connection successfull without encryption.
Retrieving user objects in LDAP directory...
Error : Could not extract users from ldap.
impacket.ldap.ldap.LDAPSearchError: Error in searchRequest -> **sizeLimitExceeded**:
Retrieving computer objects in LDAP directory...
Error : Could not extract computers from ldap.
OSError: [Errno 9] Bad file descriptor
Creating structure folders to store blob and mkf...
Connnecting to all computers to test user creds existence...
No MKF have been decrypted.
Blobs will not be decrypted.

**Comment by Processus-Thief** [comment]
Yes effectivement on m'a remonté le problème, la limite par défaut d'une requête ldap est de 1000 résultats donc ton annuaire doit avoir + de 1000 users 😉
Je vais voir pour résoudre le soucis, en attendant tu peux débloquer en ajoutant une limite plus haute dans la requête :

try : 
ldap_users = http://ldapConnection.search(searchFilter=searchFilter, attributes=['sAMAccountName', 'objectSID'], sizeLimit=9999)
except ldap.LDAPSearchError as e:
if e.getErrorString().find('sizeLimitExceeded') >=0 :
print(e) 
ldap_users = e.getAnswers()
pass
else :
raise

**Comment by kal-u** [comment]
Au top. Ta réactivité est impressionnante !
Bravo pour les nouvelles fonctionnalités,  j'allais t'en parler. Je pense à d'autres améliorations. Je vais creuser et je reviendrai vers toi ;)
